code/game/machinery/accounting.dm

```diff
@@ -0,0 +1,71 @@
+/obj/machinery/accounting
+	name = "account registration device"
+	desc = "A machine that allows heads of staff to create a new bank account after inserting an ID."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "recharger"
+	circuit = /obj/item/circuitboard/machine/accounting
+	pass_flags = PASSTABLE
+	req_one_access = list(ACCESS_HEADS, ACCESS_CHANGE_IDS)
+	var/obj/item/card/id/inserted_id
+
+/obj/machinery/accounting/Destroy()
+	. = ..()
+	if(inserted_id)
+		remove_card()
```

Comment:
Hell, I missed this one.
`Destroy()` should `return ..()` at the end of it, not call parent before.